src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect3DatasetEngine.java

```diff
@@ -132,34 +127,42 @@ public void addData(final ReferenceContext ref, final VariantContext vc, Optiona
         final int normalDepth = (int) MathUtils.sum(normalADs);
         final boolean hasNormal = normalDepth > 0;
 
+        final List<Allele> allRefAlleles = new ArrayList<>();
+        allRefAlleles.add(vc.getReference());
+        truthVCs.ifPresent(vcs -> vcs.forEach(var -> allRefAlleles.add(var.getReference())));
+        final Allele longestRef = allRefAlleles.stream().sorted(Comparator.comparingInt(Allele::length).reversed()).findFirst().get();
+
+        final List<Allele> remappedAltAlleles = ReferenceConfidenceVariantContextMerger.remapAlleles(vc, longestRef).stream()
+                .skip(1).toList();
```

Comment:
Nit:  Is `skip(1)` just because we need to skip the reference allele?  If so, just add a comment that you are making this assumption.

Reply:
done